react-delegationdashboard/src/helpers/useDelegation.ts

```diff
@@ -1,8 +1,47 @@
+import { TransactionHash } from '@elrondnetwork/erdjs/out';
 import { useContext } from 'context';
 import { Delegation } from 'contracts';
+import { DelegationTransactionType } from './contractDataDefinitions';
+import { ledgerErrorCodes } from './ledgerErrorCodes';
+export interface UseDelegationType {
+  handleClose: (txHash: TransactionHash) => void;
+  setLedgerDataError: (flag: string) => void;
+}
+export default function useDelegation({ handleClose, setLedgerDataError }: UseDelegationType) {
+  const { dapp, delegationContract, account, networkConfig } = useContext();
+  const delegation = new Delegation(dapp.proxy, delegationContract, dapp.provider, account);
+
+  const sendTransaction = (transactionArguments: DelegationTransactionType) => {
+    transactionArguments.chainId = networkConfig.chainId;
+    delegation
+      .sendTransaction(transactionArguments)
+      .then(transaction => {
+        handleClose(transaction.hash);
+      })
+      .catch(e => {
+        if (e.statusCode in ledgerErrorCodes) {
+          setLedgerDataError((ledgerErrorCodes as any)[e.statusCode].message);
+        }
+        if (e.message === 'HWApp not initialised, call init() first')
+          setLedgerDataError('Your session has expired please go and login again');
+        console.error(`${transactionArguments.type}`, e);
+      });
```

Comment:
would it worth to have an error message like `Unknown error with code X` in case none of these 2 if's happen ?